src/Compilers/CSharp/Test/Syntax/Parsing/ParserRegressionTests.cs

```diff
@@ -217,8 +217,8 @@ public void LexerTooManyBadTokens_LongUnicode()
             var parsedTree = ParseWithRoundTripCheck(source.ToString());
             IEnumerable<Diagnostic> actualErrors = parsedTree.GetDiagnostics();
             Assert.Equal("202", actualErrors.Count().ToString());
-            Assert.Equal("(1,2001): error CS1056: Unexpected character '\\U0000003C'", actualErrors.ElementAt(200).ToString());
-            Assert.Equal("(1,2011): error CS1056: Unexpected character '\\u003E\\u003E\\u003E\\u003E'", actualErrors.ElementAt(201).ToString());
+            Assert.Equal($"(1,2001): error CS1056: {string.Format(CSharpResources.ERR_UnexpectedCharacter, "\\U0000003C")}", actualErrors.ElementAt(200).ToString());
```

Comment:
>ToString() [](start = 153, length = 10)

Same as above, call IFormattable.ToString passing EnsureEnglishUICulture.PreferredOrNull as the culture, instead.